Sources/BraveWallet/Crypto/Stores/AssetDetailStore.swift

```diff
@@ -93,63 +132,118 @@ class AssetDetailStore: ObservableObject {
     walletService.defaultBaseCurrency { [self] currencyCode in
       self.currencyCode = currencyCode
     }
+    
+    fetchSupportedBuyTokens()
   }
 
   private let percentFormatter = NumberFormatter().then {
     $0.numberStyle = .percent
     $0.maximumFractionDigits = 2
   }
   
+  private func fetchSupportedBuyTokens() {
+    Task { @MainActor in
+      let buyOptions: [BraveWallet.OnRampProvider]
+      if Locale.preferredLanguages.first?.caseInsensitiveCompare("en-us") == .orderedSame {
+        buyOptions = [.ramp, .sardine, .transak]
+      } else {
+        buyOptions = [.ramp, .transak]
+      }
+      let selectedCoin = await walletService.selectedCoin()
+      selectedNetwork = await rpcService.network(selectedCoin)
+      allBuyTokensAllOptions = await blockchainRegistry.allBuyTokens(in: selectedNetwork, for: buyOptions)
+    }
+  }
+  
   public func update() {
     Task { @MainActor in
       self.isLoadingPrice = true
       self.isLoadingChart = true
-      let coin = token.coin
-      let allNetworks = await rpcService.allNetworks(coin)
-      let selectedNetwork = await rpcService.network(coin)
-      let network = allNetworks.first(where: { $0.chainId == token.chainId }) ?? selectedNetwork
-      self.network = network
-      let rampBuyTokens = await blockchainRegistry.buyTokens(.ramp, chainId: network.chainId)
-      let sardineBuyTokens = await blockchainRegistry.buyTokens(.sardine, chainId: network.chainId)
-      let buyTokens = rampBuyTokens + sardineBuyTokens
-      self.isBuySupported = buyTokens.first(where: { $0.symbol.caseInsensitiveCompare(token.symbol) == .orderedSame }) != nil
-      self.isSwapSupported = await swapService.isSwapSupported(token.chainId)
-      
-      // fetch accounts
-      let keyring = await keyringService.keyringInfo(coin.keyringId)
-      var updatedAccounts = keyring.accountInfos.map {
-        AccountAssetViewModel(account: $0, decimalBalance: 0.0, balance: "", fiatBalance: "")
-      }
       
-      if !token.isErc721 && !token.isNft {
-        // fetch prices for the asset
-        let (_, prices) = await assetRatioService.price([token.assetRatioId], toAssets: [currencyFormatter.currencyCode, "btc"], timeframe: timeframe)
-        self.isLoadingPrice = false
-        self.isInitialState = false
-        if let assetPrice = prices.first(where: { $0.toAsset.caseInsensitiveCompare(self.currencyFormatter.currencyCode) == .orderedSame }),
-           let value = Double(assetPrice.price) {
-          self.assetPriceValue = value
-          self.price = self.currencyFormatter.string(from: NSNumber(value: value)) ?? ""
-          if let deltaValue = Double(assetPrice.assetTimeframeChange) {
-            self.priceIsDown = deltaValue < 0
-            self.priceDelta = self.percentFormatter.string(from: NSNumber(value: deltaValue / 100.0)) ?? ""
+      switch assetDetailType {
+      case .blockchainToken(let token):
```

Comment:
Could we separate these two case bodies into separate functions? It's a bit hard to read and we don't have any unit test for this function. We can address unit test for existing functionality later, but perhaps we can unit test the new `AssetDetailType.coinMarket` case body function now?

Reply:
Checking further I think there is some duplicate/similar code between the two different `AssetDetailType` flows that can be shared between these two flows. 

For example the block of code that assigns `btcRatio` is duplicated between the two `AssetDetailType` cases. When fetching of price history for the chart, the only difference between the two cases is the `id` passed in, this could be separated to a shared function for re-use or even moved below the switch statement entirely and use a computed property on the `AssetDetailType` enum to return the `.assetRatioId` (blockchain token) or `.id` (coin market).